### PR TITLE
Add Unit Tests for NuveAPI

### PR DESCRIPTION
**Pull request by jcague**

### CODE REVIEW

File: nuve/nuveAPI/test/mdb/tokenRegistry.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  it('should return false if the token is not found when hasToken is called', function() {
+    var callback = sinon.stub();
+    dataBase.db.tokens.findOne.callsArgWith(1, null, undefined);
+    tokenRegistry.hasToken(kArbitraryTokenId, callback);
+
+    expect(callback.calledWith(false)).to.be.true;  // jshint ignore:line
+  });
+
+  it('should return true if the token is found in database when hasToken is called', function() {
+    var callback = sinon.spy();
```
Comment (lodoyun) [inline nuve/nuveAPI/test/mdb/tokenRegistry.js]: Consider using either spy or stub consistently. I see nothing wrong with using stubs where spies would suffice but randomly using one or the other creates confusion

File: nuve/nuveAPI/test/mdb/tokenRegistry.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  });
+
+  it('should call remove on Database when removeToken is called and it exists', function() {
+    dataBase.db.tokens.findOne.callsArgWith(1, null, undefined);
+    tokenRegistry.removeToken(kArbitraryTokenId);
+
+    expect(dataBase.db.tokens.remove.called).to.be.false;  // jshint ignore:line
+  });
+
+  it('should return true if the token is found when hasToken is called', function() {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/mdb/tokenRegistry.js]: Do you mean when removeToken is called?

File: nuve/nuveAPI/test/mdb/tokenRegistry.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    dataBase.db.tokens.findOne.callsArgWith(1, null, kArbitraryToken);
+    tokenRegistry.removeToken(kArbitraryTokenId);
+
+    expect(dataBase.db.tokens.remove.called).to.be.true;  // jshint ignore:line
+  });
+
+  it('should remove a list of tokens when getList is called', function() {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/mdb/tokenRegistry.js]: Is is 'when removeOldTokens is called' ?

File: nuve/nuveAPI/test/resource/serviceResource.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+'use strict';
+var mocks = require('../utils');
+var request = require('supertest');
+var express = require('express');
+var bodyParser = require('body-parser');
+var expect  = require('chai').expect;
+
+var kArbitraryService = {'_id': '1', rooms: []};
+
+describe('Room Resource', function() {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/resource/serviceResource.js]: Service Resource

File: nuve/nuveAPI/test/resource/serviceResource.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      request(app)
+        .get('/services/1')
+        .expect(404, 'Service not found')
+        .end(function(err) {
+          if (err) throw err;
+          done();
+        });
+    });
+
+    it('should success if service exists', function(done) {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/resource/serviceResource.js]: success -> succeed

File: nuve/nuveAPI/test/resource/servicesResource.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+'use strict';
+var mocks = require('../utils');
+var request = require('supertest');
+var express = require('express');
+var bodyParser = require('body-parser');
+var expect  = require('chai').expect;
+
+var kArbitraryService = {'_id': '1', rooms: []};
+
+describe('Room Resource', function() {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/resource/servicesResource.js]: Room Resource -> Services Resource

File: nuve/nuveAPI/test/resource/tokensResource.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+var mocks = require('../utils');
+var request = require('supertest');
+var express = require('express');
+var bodyParser = require('body-parser');
+
+var kArbitraryRoom = {'_id': '1', name: '', options: {p2p: true, data: ''}};
+var kArbitraryService = {'_id': '1', rooms: [kArbitraryRoom]};
+var kArbitraryErizoController = {hostname: 'hostname', ssl: true, ip: '127.0.0.1', port: 3000};
+
+describe('Room Resource', function() {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/resource/tokensResource.js]: Room Resource -> Tokens Resource

File: nuve/nuveAPI/test/resource/tokensResource.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        .post('/rooms/1/tokens')
+        .expect(404, 'No Erizo Controller found')
+        .end(function(err) {
+          if (err) throw err;
+          done();
+        });
+    });
+
+
+    it('should success if service and room exists', function(done) {
```
Comment (lodoyun) [inline nuve/nuveAPI/test/resource/tokensResource.js]: success -> succeed

**Comment by jcague** [comment]
@lodoyun thanks for the comments!

**Pull request merged**